### PR TITLE
[IMP] im_livechat: remove passing thread as parameter of widget PublicLivechatView/render

### PR DESCRIPTION
**Pull request by alexkuhn**
Task-2964441

### CODE REVIEW

File: addons/im_livechat/static/src/legacy/widgets/public_livechat_view/public_livechat_view.xml [hunk 1]
```diff
@@ -10,7 +10,11 @@
         @param {Array} [options.domain=[]] the domain to restrict messages on the thread.
     -->
     <t t-name="im_livechat.legacy.mail.widget.Thread">
+<<<<<<< HEAD
```
Comment (oomsveta) [inline addons/im_livechat/static/src/legacy/widgets/public_livechat_view/public_livechat_view.xml:13]: bad rebase

**Review by oomsveta — CHANGES_REQUESTED** [review]

File: addons/im_livechat/static/src/legacy/widgets/public_livechat_view/public_livechat_view.xml (hunk 1 above)
Comment (alexkuhn) [reply addons/im_livechat/static/src/legacy/widgets/public_livechat_view/public_livechat_view.xml]: change done

File: addons/im_livechat/static/src/legacy/widgets/public_livechat_view/public_livechat_view.xml [hunk 2]
```diff
@@ -10,7 +10,7 @@
         @param {Array} [options.domain=[]] the domain to restrict messages on the thread.
     -->
     <t t-name="im_livechat.legacy.mail.widget.Thread">
-        <t t-if="thread.widget.hasMessages()">
+        <t t-if="widget.messaging.publicLivechatGlobal.publicLivechat.widget.hasMessages()">
```
Comment (oomsveta) [inline addons/im_livechat/static/src/legacy/widgets/public_livechat_view/public_livechat_view.xml:13]: Is `widget` different from `widget.messaging.publicLivechatGlobal.publicLivechat.widget`? If so, is there a reason not to define `widget` to be `widget.messaging.publicLivechatGlobal.publicLivechat.widget` directly?

Comment (alexkuhn) [reply addons/im_livechat/static/src/legacy/widgets/public_livechat_view/public_livechat_view.xml:13]: The convention in old code is to access context of the widget with `widget`. OWL merges template context with component, hence why we don't need `comp` or something.

`widget !== widget.messaging.publicLivechatGlobal.publicLivechat.widget`

**Review by oomsveta — APPROVED** [review]

**Comment by alexkuhn** [comment]
Merged in grouped PR: https://github.com/odoo/odoo/pull/99137

**Pull request closed without merging**